flang/include/flang/Optimizer/CodeGen/FIROpPatterns.h

```diff
@@ -0,0 +1,249 @@
+//===-- FIROpPatterns.h -- FIR operation conversion patterns ----*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef FORTRAN_OPTIMIZER_CODEGEN_FIROPPATTERNS_H
+#define FORTRAN_OPTIMIZER_CODEGEN_FIROPPATTERNS_H
+
+#include "flang/Optimizer/CodeGen/TypeConverter.h"
+#include "mlir/Conversion/LLVMCommon/Pattern.h"
+#include "mlir/Dialect/LLVMIR/LLVMDialect.h"
+
+namespace fir {
+
+struct FIRToLLVMPassOptions;
+
+// TODO: This should really be recovered from the specified target.
```

Comment:
I am not sure the default address space is target dependent (maybe it is defined in some LLVM header though). I think the TODO comment applied to the default alignment.

Reply:
Right, `0` is always the default address space in LLVM.  Address spaces for specific objects (e.g. global variables) may be dependent on the target.

Reply:
I copied blindly the comment from the file and you are right that this does not apply for this line. I'll remove it.